core/commands/remotepin.go

```diff
@@ -0,0 +1,515 @@
+	},
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		psCh, errCh, err := lsRemote(ctx, req, env)
+		if err != nil {
+			return err
+		}
+
+		for ps := range psCh {
+			if err := res.Emit(&RemotePinOutput{
+				RequestID: ps.GetRequestId(),
+				Name:      ps.GetPin().GetName(),
+				Delegates: multiaddrsToStrings(ps.GetDelegates()),
+				Status:    ps.GetStatus().String(),
+				Cid:       ps.GetPin().GetCid().String(),
+			}); err != nil {
+				return err
+			}
+		}
+
+		return <-errCh
+	},
+	Type: RemotePinOutput{},
+	Encoders: cmds.EncoderMap{
+		cmds.Text: cmds.MakeTypedEncoder(func(req *cmds.Request, w io.Writer, out *RemotePinOutput) error {
+			fmt.Printf("pin_id=%v\n", out.RequestID)
+			fmt.Printf("pin_name=%q\n", out.Name)
+			for _, d := range out.Delegates {
+				fmt.Printf("pin_delegate=%v\n", d)
+			}
+			fmt.Printf("pin_status=%v\n", out.Status)
+			fmt.Printf("pin_cid=%v\n", out.Cid)
+			return nil
+		}),
+	},
+}
+
+func lsRemote(ctx context.Context, req *cmds.Request, env cmds.Environment) (chan pinclient.PinStatusGetter, chan error, error) {
+	opts := []pinclient.LsOption{}
+	if name, nameFound := req.Options[pinNameOptionName].(string); nameFound {
+		opts = append(opts, pinclient.PinOpts.FilterName(name))
+	}
+	if cidsRaw, cidsFound := req.Options[pinCIDsOptionName].([]string); cidsFound {
+		parsedCIDs := []cid.Cid{}
+		for _, rawCID := range cidsRaw {
+			parsedCID, err := cid.Decode(rawCID)
+			if err != nil {
+				return nil, nil, fmt.Errorf("CID %s cannot be parsed (%v)", rawCID, err)
+			}
+			parsedCIDs = append(parsedCIDs, parsedCID)
+		}
+		opts = append(opts, pinclient.PinOpts.FilterCIDs(parsedCIDs...))
+	}
+	if statusRaw, statusFound := req.Options[pinStatusOptionName].([]string); statusFound {
+		parsedStatuses := []pinclient.Status{}
+		for _, rawStatus := range statusRaw {
+			if pinclient.Status(rawStatus).String() == string(pinclient.StatusUnknown) {
+				return nil, nil, fmt.Errorf("status %s is not valid", rawStatus)
+			}
+			parsedStatuses = append(parsedStatuses, pinclient.Status(rawStatus))
+		}
+		opts = append(opts, pinclient.PinOpts.FilterStatus(parsedStatuses...))
+	}
+
+	service, _ := req.Options[pinServiceNameOptionName].(string)
+	url, key, err := getRemotePinServiceOrEnv(env, service)
+	if err != nil {
+		return nil, nil, err
+	}
+	c := pinclient.NewClient(url, key)
+
+	psCh, errCh := c.Ls(ctx, opts...)
+
+	return psCh, errCh, nil
+}
+
+var rmRemotePinCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Remove pinned objects from remote pinning service.",
+		ShortDescription: `
+Removes the pin from the given object allowing it to be garbage
+collected if needed.
+`,
+	},
+
+	Arguments: []cmds.Argument{
+		cmds.StringArg("request-id", false, true, "Request ID of the pin to be removed.").EnableStdin(),
+	},
+	Options: []cmds.Option{
```

Comment:
There was earlier input, maybe from @lidel (not sure), that the desired flag name is --request-id.

Reply:
- Single element `--id` sgtm, it maps to a single API call anyway: `DELETE /pins/{pinID}`, no need to complicate if if dont have to.
- The tricky part is `--cid=Qm1,Qm2`, which takes an array, but even if a single CID is passed may return multiple pin objects
  - due to multiple results, a single  `pin remote rm` requires multiple API calls: `GET /pins?cid=Qm1,Qm2` to get pinIDs and then `DELETE /pins/{pinID-X}` for each result.

- As for the "bulk/batch" flag, perhaps `--all` ?
  `ipfs pin remote rm --cid=Qm --all`  looks pretty self-explanatory

Reply:
I am skeptical about the definition of --force. What is the use case?

It makes sense to protect operations that would delete more than the user can see, e.g.
if we supported globbing, you want "rm *" to stop you.

But in or case the use explicitly lists the multiple pin IDs that must be deleted. So they see them in front of them.
So having to re-confirming seems a bit unncessary. (There is no implied information here as in the glob case.)

Reply:
> But in our case the user explicitly lists the multiple pin IDs that must be deleted.

That's one option that can be supported. However, another is basically a shortcut for something similar to `ipfs pin remote ls --options | ipfs pin remote rm` (we'd need to only output request-IDs for this to work). For example, we can do `ipfs pin remote rm --name="name 1"` and that could delete multiple pins.

@lidel had previously suggested the error message telling you about `--force` informing you that you could also just pass a set of request-IDs manually instead of `--force`. I recommended:

> This might be a bit overkill and perhaps we can just have the pinID be a single element instead of an array.

But I'm fine with allowing an array to get passed as well